docs/authentication.md

```diff
@@ -0,0 +1,184 @@
+
+## Register or login
+
+Before you login a user they must first be registered. We recommend you setup your code to login the user and fallback to register the account if one does not exist. This pattern is shown in the [device](#device) section.
+
+!!! Tip
+    The authentication system is very flexible. You could register a user with an email address, [link](#link-or-unlink) their Facebook account, and use it to login from another device.
+
+For a __full example__ on the best way to handle register and login in each of the clients have a look at their guides.
+
+### Device
+
+A device identifier can be used as a way to unobtrusively register a device with the server. This offers a frictionless user experience but can be unreliable because device identifiers can sometimes be changed in device updates.
+
+A device identifier must contain alphanumeric characters with dashes and be between 10 and 60 bytes.
+
+```csharp fct_label="Unity"
+var sessionHandler = delegate(INSession session) {
+  Debug.LogFormat("Session: '{0}'.", session.Token);
+};
+
+var id = PlayerPrefs.GetString("nk.id");
+if (string.IsNullOrEmpty(id)) {
+  id = SystemInfo.deviceUniqueIdentifier;
+  PlayerPrefs.SetString("nk.id", id); // cache device id.
+}
+
+var message = NAuthenticateMessage.Device(id);
+client.Login(message, sessionHandler, (INError err) => {
+  if (err.Code == ErrorCode.UserNotFound) {
+    client.Register(message, sessionHandler, (INError err) => {
+      Debug.LogErrorFormat("Error: code '{0}' with '{1}'.", err.Code, err.Message);
+    });
+  } else {
+    Debug.LogErrorFormat("Error: code '{0}' with '{1}'.", err.Code, err.Message);
+  }
+});
+```
+
+In games it is often a better option to use [Google](#google) or [Game Center](#game-center) to unobtrusively register the user.
+
+### Email
+
+Users can be registered by email and password. The password is hashed before it's stored in the database server and cannot be read or "recovered" by administrators. This protects a user's privacy.
+
+An email address must be valid as defined by RFC-5322 and passwords must be at least 8 characters.
+
+```csharp fct_label="Unity"
+string email = "email@example.com"
+string password = "3bc8f72e95a9"
+
+var message = NAuthenticateMessage.Email(email, password);
+client.Register(message, (INSession session) => {
+  Debug.LogFormat("Session: '{0}'.", session.Token);
+}, (INError err) => {
+  Debug.LogErrorFormat("Error: code '{0}' with '{1}'.", err.Code, err.Message);
+});
+// Use client.Login(...) after register.
+```
+
+### Social providers
+
+The server supports a lot of different social services with register and login. With each provider the user account will be fetched from the social service and used to setup the user. In some cases a user's [friends](social-friends.md) will also be fetched and pre-configured as part of their friends list.
```

Comment:
Is it worth linking or mentioning the social graph here? since you reference that it can be pre-configured in some cases ? just an idea.